src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveIRPass.cs

```diff
@@ -16,15 +16,31 @@ internal class DefaultDirectiveIRPass : IRazorIRPass
 
         public DocumentIRNode Execute(RazorCodeDocument codeDocument, DocumentIRNode irDocument)
         {
-            var walker = new DirectiveWalker();
-            walker.VisitDefault(irDocument);
+            var syntaxTree = codeDocument.GetSyntaxTree();
+            if (syntaxTree == null)
```

Comment:
Now might be a good time to introduce that base class we decided against earlier.